### PR TITLE
Remove "query" query and fix related parsing bugs

### LINKED ISSUE #13326: The delete-by-query plugin wraps all queries in a `query` element
The delete-by-query plugin wraps all queries in a `query` element, so the parsed queries are wrapperd in a QueryWrapperFilter in the end. This does not change the matching documents but prevents me from removing the `query` query which is deprecated (which takes a query and turns it into a filter).

### PR DESCRIPTION
**Pull request by javanna**
We have two types of parse methods for queries: one for the inner query, to be used once the parser is positioned within the query element, and one for the whole query source, including the query element that wraps the actual query.

With the search refactoring we ended up using the former in count, cat count and delete by query, whereas we should have used the former.  It ends up working properly given that we have a registered (deprecated) query called "query", which used to allow to wrap a filter into a query, but this has the following downsides:
1) prevents us from removing the deprecated "query" query
2) we end up supporting a top level query that is not wrapped within a query element (pre 1.0 syntax iirc that shouldn't be supported anymore)

This commit finally removes the "query" query and fixes the related parsing bugs. We also had some tests that were providing queries in the wrong format, those have been fixed too.

Closes #13326

### CODE REVIEW

File: docs/reference/migration/migrate_3_0.asciidoc [hunk 1]
```diff
@@ -78,6 +78,16 @@ format can't read from version `3.0.0` and onwards. The new format allows for a
 scalable join between parent and child documents and the join data structures are stored on on disk
 data structures as opposed as before the join data structures were stored in the jvm heap space.
 
+==== Deprecated queries removed
+
+The following deprecated queries have been removed:
+* `filtered`: use `bool` query instead, which supports `filter` clauses too
+* `and`: use `must` clauses in a ``bool` query instead
```
Comment (jpountz) [inline docs/reference/migration/migrate_3_0.asciidoc]: s/``bool`/`bool`/

File: docs/reference/migration/migrate_3_0.asciidoc [hunk 2]
```diff
@@ -78,6 +78,16 @@ format can't read from version `3.0.0` and onwards. The new format allows for a
...
+* `or`: use should clauses in a `bool` query instead
+* `limit`: use `terminate_after` parameter instead
+* `fquery`: obsolete after filters and queries have been deprecated
```
Comment (jpountz) [inline docs/reference/migration/migrate_3_0.asciidoc]: s/deprecated/merged/ ?

**Comment by jpountz** [comment]
This looks great! I didn't know we supported calling the `_search` API and providing the query without nesting it into a `query` object. I'm wondering if we want to keep supporting that... (out of the scope of this PR)

**Comment by javanna** [comment]
> I didn't know we supported calling the _search API and providing the query without nesting it into a query object

We don't, we used to pre 1.0 I think, but maybe my explanation above was confusing. I just meant that depending on where we are in the parsing process, we use one of the two methods, think of the difference between count for instance, which only supports a query element, and search, which supports all the `SearchParseElement`s (`aggs`, `size` etc.). Hope this clarifies things. Thanks for the review!

**Comment by jpountz** [comment]
ok, great!

**Pull request closed without merging**